AbstractRecurrent.lua

```diff
@@ -40,7 +41,8 @@ end
 
 function AbstractRecurrent:trimZero(nInputDim)
    self.recurrentModule = nn.TrimZero(self.recurrentModule, nInputDim, true)
-   self.sharedClones = {self.recurrentModule}
+   self.sharedClones = {self.recurrentModule:stepClone()}
```

Comment:
This should be `self.sharedClones = {self.recurrentModule}` or it will cause issues with BatchNormalization and such.